src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorLanguageServer.cs

```diff
@@ -241,9 +241,9 @@ public static Task<RazorLanguageServer> CreateAsync(Stream input, Stream output,
 
                         // Completion
                         services.AddSingleton<CompletionListCache>();
-                        services.AddSingleton<AggregateCompletionListProvider>();
-                        services.AddSingleton<CompletionListProvider, DelegatedCompletionListProvider>();
-                        services.AddSingleton<CompletionListProvider, RazorCompletionListProvider>();
+                        services.AddSingleton<CompletionListProvider>();
```

Comment:
Yeah I agree with this abstraction, it will probably end up mattering where the providers were sourced from so they should be named specific things.